### ISSUE TITLE
QDef-in-InstMod Unmerged Commits

### ISSUE THREAD

**Issue by Mornix** [JacquesCarette/Drasil#1373]
There are five commits which exist on `QDef-in-InstMod` which did not make the merge in #1148. 

1. "Instance Models now are divided into Equational, DE, and Other. Only Other are built ATM, but the next pass will refine that. This does eliminate the relToQD hack." (560d776)

Things you did in the commit:
* Gave `QDefinition` a `ConceptDomain`. (All constructors set it as empty.)
* Replaced `RelationConcept` in `InstanceModel` with `ModelKind`; made a `Lens` for `ModelKind`. ([Definition](https://github.com/JacquesCarette/Drasil/commit/560d77699d707d2aa2f59baa9f37e25a11ad0c3f#diff-971b886abccf263018d10dbd52907c7fR40) — not a permalink)
* Remove `relToQD` and `convertRel` and replaced it with `getEqMod` ([definition](https://github.com/JacquesCarette/Drasil/commit/560d77699d707d2aa2f59baa9f37e25a11ad0c3f#diff-971b886abccf263018d10dbd52907c7fR107) — not a permalink)
* Remove `Lens` from the typeclass `ExprRelat`, adding the comment ["It is ok to be able to view a (defining?) 'Relation', but not necessarily to 'set' it,  as it might just not be settable. So Getter it is."](https://github.com/JacquesCarette/Drasil/commit/560d77699d707d2aa2f59baa9f37e25a11ad0c3f#diff-dc79e4fa0265786b62fda39ecdd4b780R126)

2. "properly use the new instance model constructors. This seems to point out all sorts of oddities. Regularized some (SSP and SWHS) and left some in (GlassBR and GamePhysics). Unclear what is really going on, need to get @smiths to take a good look." (f23bf7b)

Things you did in the commit:
* Replaced many `RelationConcept`s in the examples with `QDefinition`.
* Added `InstanceModel` smart constructors to create a specific kind of `InstanceModel`. ([Definition](https://github.com/JacquesCarette/Drasil/commit/f23bf7b5757b6686d4fa1a6aec371520300ee77b#diff-971b886abccf263018d10dbd52907c7fR94) — not a permalink)
* Swapped many `InstanceModel`s from using the `im'` smart constructor to using new ones creating specific instantiations of `ModelKind`s. 

* Added a few FIXMEs to equations:
  * [FIXME: the 'equation' used to be a sequence of equations](https://github.com/JacquesCarette/Drasil/commit/f23bf7b5757b6686d4fa1a6aec371520300ee77b#diff-7e4cef38e330b3b1229b2c4bd46f6479R36)
  * [FIXME: othModel because it should be a Functional Model](https://github.com/JacquesCarette/Drasil/commit/f23bf7b5757b6686d4fa1a6aec371520300ee77b#diff-7e4cef38e330b3b1229b2c4bd46f6479R92)
  * [FIXME, see comment below](https://github.com/JacquesCarette/Drasil/commit/f23bf7b5757b6686d4fa1a6aec371520300ee77b#diff-8f9869700a06c62936711c1647e90d6eR31) with the comment below being "`calofDemandDesc $ (C demand) $= FCall (asExpr interpY) [V "TSD.txt", sy standOffDist, sy eqTNTWeight] `"
  * [FIXME: this is an 'other' model because we don't have Functional models yet](https://github.com/JacquesCarette/Drasil/commit/f23bf7b5757b6686d4fa1a6aec371520300ee77b#diff-9de3590e3f6541a739713d985f90b267R359)

3. "_definitions in SystemInformation were indeed no longer used." (39f2c22)

Things you did in the commit:
* Removed `_definition` from `SystemInformation` and removed everywhere it propagated to. Not sure if this is still immediately possible or a good idea as `defs'` is now used in `CodeSpec.hs` (as of be8f757) where it was previously unused. I'm going to /cc @bmaclach here since he knows the codegen better than I do.

4. "making name of variables and display consistent, i.e. use j rather than i a subscript." (266a480)

Things you did in the commit:
* Rename `mass_i` and `pos_i` in game physics to be `_j` and propagate that through the example.

5. "latex was broken -- dmath was being used, but package breqn had been disabled. Switched back to displaymath instead. Can go back to dmath once it is debugging with lualatex." (7c211cf)

This commit wasn't merged with #1148 because the changes were already made separately in `master` by that point to get TeX compiling.

**Comment by JacquesCarette** [comment]
Perfect, thank you!

**Comment by bmaclach** [comment]
The `_definitions` field is how the code generator knows about the `InstanceModel`s, so removing it now would indeed break the generated code.

**Comment by JacquesCarette** [comment]
Ok, I won't!  When I first did this, apparently things still worked, or maybe were insufficiently tested, so I did not notice. Thanks @Mornix and @bmaclach for 'doing the right thing' on the comments on the issues.

**Comment by JacquesCarette** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [QDef-in-InstMod](https://trello.com/c/IV0b87F1/646-qdef-in-instmod)

**Comment by smiths** [comment]
@JacquesCarette, it looks like you included a link in the above to your Trello board.  If you intention was for others to see the board, it doesn't work because of a card cannot be found error, likely because your board is private.

**Comment by JacquesCarette** [comment]
I meant to include a link to this issue on my Trello board (which worked). As side-effect of doing that seems that the reverse also happens, which I did not expect! And indeed, it is private (and I won't change that!). Hmm, I wonder if I can turn that reverse-link thing off.

**Comment by samm82** [comment]
I don't think I'll be able to fix my branch by tomorrow; I'm still having the issue where the name of the quantity output by the IMs is being taken from the name of the IM. Should I keep the branch open or should I delete it? I don't think the work that I did is intrinsically valuable, as I just manually went through the `QDef-in-InstMod` branch and started merging the earliest commits (I didn't get very far!) and the longer this branch stays open, the farther behind `master` it becomes. @JacquesCarette
